python/cudf/cudf/core/frame.py

```diff
@@ -299,10 +300,33 @@ def _copy_categories(self, other, include_index=True):
         return self
 
     def _unaryop(self, op):
-        result = self.copy()
-        for name, col in result._data.items():
-            result._data[name] = col.unary_operator(op)
-        return result
+        data_columns = (col.unary_operator(op) for col in self._columns)
+        data = OrderedColumnDict(zip(self._column_names, data_columns))
```

Comment:
Technically `data` here can be a plain dict and the `Frame` constructor will handle converting it to the appropriate  mapping type (`ColumnAccessor`).

Reply:
Sorry - I realize I wasn't making a suggestion in the above comment: let's just use a plain `dict` here, and keep `OrderedColumnDict` an internal detail to `Frame`/`Table`.